apps/src/blockly/addons/functionEditor.js

```diff
@@ -370,4 +372,33 @@ export default class FunctionEditor {
     this.editorWorkspace.clear();
     Blockly.Events.enable();
   }
+
+  // If the definition block has a description property
+  // under definitionBlock.description, move it to the description field
+  // of the block, and delete the description property from the definition block.
+  // If the block does not have a description field, don't move the description.
+  moveDescriptionToBlockField(definitionBlock) {
+    if (
```

Comment:
nit: If you're willing to use lodash, maybe this makes it clearer what you're doing?

```
const DESCRIPTION_INPUT_PATH = 'inputList[1].fieldRow[1]';

moveDescriptionToBlockField(definitionBlock) {
  const descriptionDestinationExists = has(definitionBlock, DESCRIPTION_INPUT_PATH);
  if (definitionBlock.description && descriptionDestinationExists) {
    // TODO: setValue is an XML-specific method, right? Will this work?
    definitionBlock.set(DESCRIPTION_INPUT_PATH, definitionBlock.description);
    definitionBlock.description = undefined;
  }

```

Reply:
good idea

Reply:
I don't know if all my null checks are strictly necessary, but it feels weird to me to assume that long chain will always exist.

Reply:
I updated this, I couldn't get `set` or `setValue` to work so I stuck with my original code there. But I am now using `_.has`.